src/prompts/index.tsx

```diff
@@ -198,6 +204,55 @@ export class PromptManager {
         return false;
     }
 
+    private maybePersistReminder(prompt: PromptConfig) {
+        if (prompt.type !== 'alert') {
+            return;
+        }
+
+        const reminder = prompt.declinedPromptReminderDelay
+            ? { promptUuid: prompt.uuid, declinedOn: Date.now() }
+            : 'suppressed';
+
+        persistPromptReminder(
+            prompt.uuid,
+            reminder
+        );
+    }
+
+    private hidePrompt(prompt: PromptConfig) {
+        const idx = this.activePrompts.indexOf(prompt);
+        this.activePrompts.splice(idx, 1);
+
+        this.render();
+    }
+
+    private async isPromptSuppressed(prompt: PromptConfig): Promise<boolean> {
+        if (prompt.type !== 'alert') {
+            return Promise.resolve(false);
+        }
```

Comment:
Necessary? If I follow, would we not fall into if (!reminder) -> false anyway for other types?

Reply:
No longer relevant as impl has changed now